tests/test_astrometryModel.py

```diff
@@ -0,0 +1,352 @@
+
+        self.associations.associateCatalogs(matchCut)
+        self.associations.prepareFittedStars(minMeasurements)
+        self.associations.deprojectFittedStars()
+
+    def _prepModels(self):
+        """Call this after model1 and model2 are created, to call assignIndices,
+        and instantiate the fitters.
+        """
+        posError = 0.02  # in pixels
+        # have to call this once or offsetParams will fail because the transfo indices aren't defined
+        self.model1.assignIndices("Distortions", self.firstIndex)
+        self.fitter1 = lsst.jointcal.AstrometryFit(self.associations, self.model1, posError)
+
+        # have to call this once or offsetParams will fail because the transfo indices aren't defined
+        self.model2.assignIndices("Distortions", self.firstIndex)
+        self.fitter2 = lsst.jointcal.AstrometryFit(self.associations, self.model2, posError)
+
+    def testMakeSkyWcsModel1(self):
+        self._testMakeSkyWcs(self.model1, self.fitter1, self.inverseMaxDiff1)
+
+    def testMakeSkyWcsModel2(self):
+        self._testMakeSkyWcs(self.model2, self.fitter2, self.inverseMaxDiff2)
+
+    def _testMakeSkyWcs(self, model, fitter, inverseMaxDiff):
+        """Test producing a SkyWcs on a model for every cdImage,
+        both post-initialization and after one fitting step.
+        """
+        # first test on as-initialized models
+        for ccdImage in self.associations.getCcdImageList():
+            self.checkMakeSkyWcs(model, ccdImage, inverseMaxDiff)
+
+        # now shift the models to non-default, but more reasonable, values by taking one fitting step.
+        fitter.minimize("DistortionsVisit")
+        fitter.minimize("Distortions")
+        for ccdImage in self.associations.getCcdImageList():
+            self.checkMakeSkyWcs(model, ccdImage, inverseMaxDiff)
+
+    def checkMakeSkyWcs(self, model, ccdImage, inverseMaxDiff):
+        """Test converting the model of one ccdImage to a SkyWcs by comparing
+        to the original transfo at the tangent plane.
+
+        Parameters
+        ----------
+        model : `lsst.jointcal.AstrometryModel`
+            The model to test.
+        ccdImage : `lsst.jointcal.CcdImage`
+            The ccdImage to extract from the model and test.
+        inverseMaxDiff : `float`
+            Required accuracy on inverse transform.
+            See `lsst.afw.geom.utils.assertPairsAlmostEqual`.
+        """
+        skyWcs = model.makeSkyWcs(ccdImage)
+        sky2TP = model.getSky2TP(ccdImage)
+        mapping = model.getMapping(ccdImage)
+        posErr = 1e-3  # need something for positional error
+
+        bbox = ccdImage.getDetector().getBBox()
+        num = 200
+        xx = np.linspace(bbox.getMinX(), bbox.getMaxX(), num)
+        yy = np.linspace(bbox.getMinY(), bbox.getMaxY(), num)
+        points = []
+        for x in xx:
+            for y in yy:
+                points.append(lsst.afw.geom.Point2D(x, y))
```

Comment:
`points = [lsst.afw.geom.Point2D(*xy) for xy in itertools.product(xx, yy)]`

Reply:
Oh, I totally forgot about itertools.product. Nice catch!